File: app/src/main/java/s15666/pjwstk/pamobmi/benedictharris/ImperialHBEquation.java
package s15666.pjwstk.pamobmi.benedictharris;

public final class ImperialHBEquation extends HarrisBenedictEquation {

    @Override
    double getGenderBaseValue(Gender gender) {
        return gender.equals(Gender.MALE) ? 66 : 655.1;
    }

    @Override
    double getWeightMultiplier(Gender gender) {
        return gender.equals(Gender.MALE) ? 6.2 : 4.35;
    }

    @Override
    double getHeightMultiplier(Gender gender) {
        return gender.equals(Gender.MALE) ? 12.7 : 4.7;
    }

    @Override
    double getAgeMultiplier(Gender gender) {
        return gender.equals(Gender.MALE) ? 6.76 : 4.7;
    }
}

File: app/src/main/java/s15666/pjwstk/pamobmi/benedictharris/ImperialHBEquation.kt
package s15666.pjwstk.pamobmi.benedictharris

class ImperialHBEquation : HarrisBenedictEquation() {

    override fun getGenderBaseValue(gender: Gender?): Double {
        return if (gender == Gender.MALE) 66.0 else 655.1
    }

    override fun getWeightMultiplier(gender: Gender?): Double {
        return if (gender == Gender.MALE) 6.2 else 4.35
    }

    override fun getHeightMultiplier(gender: Gender?): Double {
        return if (gender == Gender.MALE) 12.7 else 4.7
    }

    override fun getAgeMultiplier(gender: Gender?): Double {
        return if (gender == Gender.MALE) 6.76 else 4.7
    }
}